src/components/flow/actions/localization/MsgLocalizationForm.tsx

```diff
@@ -320,22 +335,66 @@ export default class MsgLocalizationForm extends React.Component<
         buttons={this.getButtons()}
         tabs={tabs}
       >
+        {renderIf(!!this.context.config.onSystemTranslate)(
+          <label className={styles.translation_label}>Original Text</label>
+        )}
         <div data-spec="translation-container">
           <div data-spec="text-to-translate" className={styles.translate_from}>
             {(this.props.nodeSettings.originalAction as SendMsg).text}
           </div>
         </div>
 
-        <TextInputElement
-          name={i18n.t('forms.message', 'Message')}
-          showLabel={false}
-          onChange={this.handleMessageUpdate}
-          entry={this.state.message}
-          placeholder={`${this.props.language.name} ${translation}`}
-          autocomplete={true}
-          focus={true}
-          textarea={true}
-        />
+        {renderIf(!!this.context.config.onSystemTranslate)(
+          <div className={styles.translate_to_container}>
+            <div className={styles.translate_to_item}>
+              <label
+                className={styles.translation_label}
+              >{`${this.props.language.name} ${translation}`}</label>
+              <TextInputElement
+                name={i18n.t('forms.message', 'Message')}
+                showLabel={false}
+                onChange={this.handleMessageUpdate}
```

Comment:
Just a question, the translation happens in real-time, right? I mean, when the state changes it calls the system to translate?

Reply:
I have added changes to call the backend with a 1sec timeout after the user last typed.

Reply:
Great! Thank you